File: BOJ/Algorithms/DFS_BFS/2606.py
import sys ; input = lambda : sys.stdin.readline().rstrip()
from collections import deque

matrix = [[0] * 101 for i in range(101)]
visitList = [False] * 101

def BFS(N : int, A : int) -> int:
    Q = deque()
    result = 0
    Q.append(A)
    visitList[A] = True
    while(Q):
        A = Q.popleft()
        for i in range(1, N+1):
             if matrix[A][i] == 1 and visitList[i] == False:
                visitList[i] = True
                Q.append(i)
                result += 1
    return result

N = int(input())
M = int(input())
for i in range(M):
    A, B = map(int, input().split())
    matrix[A][B] = matrix[B][A] = 1
Answer = BFS(N, 1)
print(Answer)

File: BOJ/Algorithms/DFS_BFS/2606.cpp
#include <iostream>
#include <deque>
#define MAX 101
using namespace std;

int matrix[MAX][MAX];
bool visitList[MAX];

int BFS(int N, int A){
    deque<int> Q;
    int result = 0;
    Q.push_back(A);
    visitList[A] = true;
    while(!Q.empty()){
        A = Q.front();
        Q.pop_front();
        for(int i = 1; i <= N; i++){
            if(matrix[A][i] && !visitList[i]){
                visitList[i] = true;
                Q.push_back(i);
                result += 1;
            }
        }
    }
    return result;
}

int main(){
    int N, M;
    cin >> N;
    cin >> M;
    for(int i = 0; i < M; i++){
        int A, B;
        cin >> A >> B;
        matrix[A][B] = 1;
        matrix[B][A] = 1;
    }
    int Answer = BFS(N, 1);
    cout << Answer << '\n';
}